docs/language/functions.md

```diff
@@ -0,0 +1,824 @@
+echo '{foo:"Zed"}' | zq -z 'foo := to_upper(foo)' -
+```
+
+**Output:**
+```mdtest-output
+{foo:"ZED"}
+```
+
+### `trim`
+
+```
+trim(s <stringy>) -> string
+```
+
+`trim` removes all leading and trailing whitespace from the string `s`.
+
+```mdtest-command
+echo '{foo:"   Zed   "}' | zq -z 'foo := trim(foo)' -
+```
+
+**Output:**
+```mdtest-output
+{foo:"Zed"}
+```
+
+## Time
+
+### `trunc`
+
+```
+trunc(t <timey>, m (duration,<number>)) -> time
+```
+
+<!-- XXX document time coercion rules  and link below-->
+
+`trunc` returns time `t` (or value that can be coerced to time) rounded down to
+the nearest multiple of duration `m`.
+
+```mdtest-command
+echo '{ts:2020-05-26T15:27:47Z}' | zq -z 'ts := trunc(ts, 1h)' -
+```
+
+**Output:**
+```mdtest-output
+{ts:2020-05-26T15:00:00Z}
+```
+
+### `len`
+
+```
+len(v (record,array,set,map,bytes,string,bstring,ip,net,error)) -> int64
+```
+
+`len` returns the length of value `v`. Supported types:
+
+- record
+- array
+- set
+- map
+- bytes
+- string
+- bstring
+- ip
+- net
+- error
+
+#### Example:
+
+```mdtest-command
+echo '{foo:[1,2,3]}' | zq -z 'foo := len(foo)' -
+```
+
+**Output:**
+```mdtest-output
+{foo:3}
+```
+
+## Types
+
+### `is`
+
+```
+is([s <any>], t type) -> bool
+```
+
+`is` returns true if the subject value s is of type t. `is` can accept either
+one arguments or two: Value `s` is optional and if omitted the subject value is
+the root value (i.e. equivalent to `is(this, type)`).
+
+#### Example:
+
+```mdtest-command
+echo '{foo:1.}' | zq -z 'foo := is(foo, float64)' -
+echo '{foo:1.}' | zq -z 'foo := is({foo:float64})' -
+```
+
+**Output:**
+```mdtest-output
+{foo:true}
+{foo:true}
+```
+
+### iserr
+
+```
+iserr(v <any>) -> bool
+```
+
+`iserr` returns true if value v is of type error.
+
+#### Example:
+
+```mdtest-command
+echo '{foo:"this is an error"(error)}' | zq -z 'foo := iserr(foo)' -
+```
+
+**Output:**
+```mdtest-output
+{foo:true}
+```
+
+### nameof
+
+```
+nameof(v <any>) -> string
+```
+
+`nameof` returns the string type name of v if v is an aliased type.
+
+### `typeof`
+
+```
+typeof(a <any>) -> type
+```
+
+`typeof` returns the [type](../formats/zson.md#357-type-type) of value a.
+
+#### Example:
+
+```mdtest-command
+echo  '{foo:127.0.0.1}' \
+  | zq -z 'foo := typeof(foo)' -
+```
+
+**Output:**
+```mdtest-output
+{foo:(ip)}
+```
+
+### `typeunder`
+
+```
+typeunder(a <any>) -> type
+```
+
+`typeunder` returns the [type](../formats/zson.md#357-type-type) of value `a`.
+`typeunder` is similar to `typeof` except that if `a` is [named type](../formats/zson.md#357-type-type)
+the type under `a` is returned.
+
+#### Example:
+
+```mdtest-command
+echo  '{flavor:"chocolate"}(=flavor)' \
+  | zq -z 'cut typeunder := typeunder(this)' -
+```
+
+**Output:**
+```mdtest-output
+{typeunder:({flavor:string})}
+```
+
+## Value Introspection
+
+### `missing`
+
+```
+missing(e <expression>) -> bool
+```
+
+`missing` returns true if a value in [expression](./expressions.md) e is
```

Comment:
Nit: We generally don't include a `./` prefix in relative links. (This applies through the file.)
```suggestion
`missing` returns true if a value in [expression](expressions.md) `e` is
```